src/main/java/roomescape/repository/ReservationDao.java

```diff
@@ -0,0 +1,90 @@
+
+@Repository
+public class ReservationDao {
+
+    public static final String RESERVATION_TABLE = "reservation";
+    public static final String RESERVATION_PK = "id";
+
+    private final JdbcTemplate jdbcTemplate;
+    private final SimpleJdbcInsert reservationInserter;
+    private final RowMapper<Reservation> reservationRowMapper = (resultSet, rowNum) ->
+            new Reservation(
+                    resultSet.getLong("reservation_id"),
+                    resultSet.getString("name"),
+                    resultSet.getDate("date").toLocalDate(),
+                    new ReservationTime(
+                            resultSet.getLong("time_id"),
+                            resultSet.getTime("time_value").toLocalTime()
+                    )
+            );
+
+    public ReservationDao(final JdbcTemplate jdbcTemplate, final DataSource dataSource) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.reservationInserter = new SimpleJdbcInsert(dataSource)
+                .withTableName(RESERVATION_TABLE)
+                .usingGeneratedKeyColumns(RESERVATION_PK);
+    }
+
+    public List<Reservation> getReservations() {
+        final String sql = """
+                SELECT
+                    r.id AS reservation_id,
+                    r.name,
+                    r.date,
+                    t.id AS time_id,
+                    t.start_at AS time_value
+                FROM reservation AS r 
+                INNER JOIN reservation_time AS t 
+                ON r.time_id = t.id
+                """;
+        return jdbcTemplate.query(sql, reservationRowMapper);
+    }
+
+    public Reservation createReservation(final Reservation convertedRequest) {
+        long id = insertReservationAndRetrieveKey(convertedRequest);
+        return getReservationById(id);
+    }
+
+    public void deleteReservationById(final long id) {
+        final String sql = "DELETE FROM reservation WHERE id = ?";
+        jdbcTemplate.update(sql, id);
+    }
+
+    private Reservation getReservationById(final long id) {
+        final String sql = """
+                SELECT
+                    r.id AS reservation_id,
+                    r.name,
+                    r.date,
+                    t.id AS time_id,
+                    t.start_at AS time_value
+                FROM reservation AS r 
+                INNER JOIN reservation_time AS t 
+                ON r.time_id = t.id
+                WHERE r.id = ?
+                """;
+        return jdbcTemplate.queryForObject(sql, reservationRowMapper, id);
+    }
+
+    private long insertReservationAndRetrieveKey(final Reservation convertedRequest) {
+        final Map<String, Object> parameters = new HashMap<>(Map.of(
+                "name", convertedRequest.getName(),
+                "date", Date.valueOf(convertedRequest.getDate()),
+                "time_id", convertedRequest.getTimeId())
+        );
+        return (long) reservationInserter.executeAndReturnKey(parameters);
```

Comment:
```suggestion
        return reservationInserter.executeAndReturnKey(parameters).longValue();
```
.longValue() 를 고려해봐도 좋을 것 같아요.

Reply:
622de64

가독성이나 안정성 측면에서 더 좋은 방식이네요!
반영했습니다 👍